src/DependencyInjection/ApplicationCompilerPass.php

```diff
@@ -60,13 +59,33 @@ public function process(ContainerBuilder $container): void
         }
 
         $language = $requestInfo->getLanguage();
-        if (!isset($langRepositories[$version][$language])) {
+        if ($language === null || !isset($langUrls[$version]['github'][$language])) {
             $language = self::FALLBACK_LANGUAGE;
         }
 
+        $controller = $requestInfo->getController();
+        if ($controller === null || !isset($mapping[$controller])) {
+            $controller = self::FALLBACK_CONTROLLER;
+        }
+
+        $providerInfo = new ProviderInfo($requestInfo, $version, $language, $controller);
+        $container->set(ProviderInfo::class, $providerInfo);
+        $container->setDefinition(ProviderInfo::class, (new Definition(ProviderInfo::class))->setSynthetic(true));
+
+        $baseSlug = implode('/', array_slice(explode('/', $mapping[$controller][$language]), 0, -1));
+        $relativeLinkUrl = !isset($langUrls[$version]['gitbook'][$language]) || !is_string($container->getParameter('docs_base_url'))
+            ? null
+            : $container->getParameter('docs_base_url') . '/' . $langUrls[$version]['gitbook'][$language] . '/' . $baseSlug
+        ;
+
         $container->setParameter('mapping', $mapping);
-        $container->setParameter('language', $language);
-        $container->setParameter('repository', $langRepositories[$version][$language]);
+        $container->setParameter('repository', $langUrls[$version]['github'][$language]);
+        $container->setParameter('markdown_config', [
```

Comment:
Maybe just pass the whole content instead of something depending of the request.